lms/static/coffee/src/instructor_dashboard/queries.coffee

```diff
@@ -0,0 +1,685 @@
+                {'text': jQuery(cells[2]).text(), 'id':cells[2].id},
+                {'text': jQuery(cells[3]).text()}]
+
+        savedQueryOptionsText = [jQuery(cells[0]).text(), jQuery(cells[1]).text(),
+                     jQuery(cells[2]).text(), jQuery(cells[3]).text()]
+        @tr = @start_row( jQuery(cells[0]).text().toLowerCase(),
+          savedQueryOptions,'', $('.emailWidget.queryTableBody'))
+        # todo:this feels too hacky. suggestions?
+        @useQueryEndpoint = [
+          @$queryEndpoint,
+          savedQueryOptionsText.slice(0,2).join('/'),
+          savedQueryOptions[2].id
+        ].join('/')
+        @filtering = savedQueryOptions[3].text
+        @entityName = savedQueryOptions[2].text
+        @reload_students(@tr)
+        @$emailListContainers.find('select.single-email-selector').
+          prop('selectedIndex',0)
+        $('.problem_specific').removeClass('active')
+        $('.section_specific').removeClass('active')
+    $td = $('<td>')
+    $td.append($loadBtn)
+    row.appendChild($td[0])
+    $deleteBtn = $(_.template('<div class="deleteSaved">
+      <i class="icon-remove-sign"></i> <%= label %></div>', {label: 'Delete'}))
+
+    $deleteBtn.click (event) =>
+      targ = event.target
+      while (!targ.classList.contains('deleteSaved'))
+        targ = targ.parentNode
+      curRow = targ.parentNode.parentNode
+      curRow.remove()
+      queryToDelete = curRow.getAttribute('groupquery')
+      @delete_saved_query(queryToDelete)
+      $('#send_email option[value="' + queryToDelete + '"]').remove()
+    $td = $('<td>')
+    $td.append($deleteBtn)
+    row.appendChild($td[0])
+    return $(row)
+
+  get_students: (cb) ->
+    tab = $('.emailWidget.queryTableBody')
+    rows = tab.find('tr')
+    _.each rows, (row) ->
+      type = row.classList[0]
+      problems = []
+      _.each row.children, (child) ->
+        id = child.id
+        html = child.innerHTML
+        problems.push({'id': id, 'text': html})
+      problems = problems.slice(0, -1)
+    send_data =
+      filter: @filtering
+      entityName: @entityName
+    $.ajax
+      dataType: 'json'
+      url: @useQueryEndpoint
+      data: send_data
+      success: (data) -> cb? null, data
+      error: std_ajax_err ->
+        cb? gettext('Error getting students')
+
+  # make a single query to the backend. doesn't wait for
+  # query completion as that can take awhile
+  reload_students: (tr) ->
+    @$saveQueryBtn.addClass('disabled')
+    @$emailCsvBtn.addClass('disabled')
+    @$emailCsvBtn[0].value = gettext('Aggregating Queries')
+    tr.addClass('working')
+    @get_students (error, students) =>
+      if error
+        $broken_icon = $ _.template('<div class="done">
+          <i class="icon-warning-sign"></i> <%= label %></div>',
+          {label: gettext("Sorry, we're having a problem with this query.
+            Please delete this row and try again.")})
+        tr.children()[4].innerHTML = $broken_icon[0].outerHTML
```

Comment:
This error message needs to be handled in clear_display. Looks like it's dynamic tho, so I don't know if it would be easy to access the div to get rid of it.